### PR TITLE
feat: add ability to use assertions

### PR DESCRIPTION
**Pull request by BaseInfinity**
I have a pull request open that outputs the generated code, I think what would make this useful in the testing space if it had the ability to assert things as well.

Let me know if this is starting to deviate too much from your original intentions. 

I forked with intentions of making this an E2E testing tool for Playwright, whereas your intention seems to be more general purpose and you may not want all of my changes so feel free to reject changes that don't match eye to eye with your goals =)

### CODE REVIEW

**Comment by BaseInfinity** [comment]
proof if it working =)

Screen Shot 2023-12-14 at 4 27 02 PM

File: actions/interactWithPage.js [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 import {parseSite, preprocessJsonInput} from '../util/index.js';
 import {HumanMessage, SystemMessage} from 'langchain/schema';
+import {expect} from '@playwright/test';
 
 const AsyncFunction = async function () {}.constructor;
 
 export async function interactWithPage(chatApi, page, task) {
   const code = await getPlayWrightCode(page, chatApi, task);
-  return execPlayWrightCode(page, code);
+  return execPlayWrightCode(page, code, expect);
```
Comment (mayt) [inline actions/interactWithPage.js:10]: I dont think you need to pass in down here

File: actions/interactWithPage.js [hunk 2]
```diff
@@ -67,10 +68,10 @@ await page.getByText(articleByText, { exact: true }).click(articleByText);
...
 }
 
-async function execPlayWrightCode(page, code) {
+async function execPlayWrightCode(page, code, expect) {
   try {
-    const func = AsyncFunction('page', code);
-    return await func(page);
+    const func = AsyncFunction('page', 'expect', code);
+    return await func(page, expect);
   } catch (e) {
```
Comment (mayt) [inline actions/interactWithPage.js:75]:
```
async function execPlayWrightCode(page, code) {
  try {
    const dependencies = [
        {param: 'page', value: page}, 
        {param: 'expect', value: expect},  
   ]
    const func = AsyncFunction(...dependencies.map(d => d.param), code);
    return await func(...dependencies.map(d => d.value));
  } catch (e) {

**Comment by BaseInfinity** [comment]
Thanks for the CR! I'll most likely get to it later tonight or tomorrow morning =)

**Comment by BaseInfinity** [comment]
Thanks for the CR, this is ready for review again

File: actions/interactWithPage.js [hunk 3]
```diff
@@ -68,9 +69,14 @@ await page.getByText(articleByText, { exact: true }).click(articleByText);
...
+  const dependencies = [
+    {param: 'page', value: page},
+    {param: 'expect', value: expect},
+  ];
+
   try {
-    const func = AsyncFunction('page', code);
-    return await func(page);
+    const func = AsyncFunction(...dependencies.map((d) => d.param), code);
+    return await func(page, expect);
```
Comment (mayt) [inline actions/interactWithPage.js:79]: Please update this line to use the dependencies too

Comment (BaseInfinity) [reply actions/interactWithPage.js]: https://github.com/mayt/BrowserGPT/pull/17/commits/7c1fe6fc96dac077cfda26dc0bb1bd3d02ceb94b

**Comment by BaseInfinity** [comment]
This is ready for more CR =)

**Review by mayt — APPROVED** [review]

**Pull request merged**